api/src/main/java/io/opencensus/trace/unsafe/ContextUtils.java

```diff
@@ -39,7 +40,35 @@ private ContextUtils() {}
    * The {@link io.grpc.Context.Key} used to interact with {@link io.grpc.Context}.
    *
    * @since 0.5
+   * @deprecated from API since 0.21. Use {@link #withValue(Context, Span)} and {@link
+   * #getValue(Context)} instead.
    */
+  // TODO(songy23): make this private once gRPC migrates to use the alternative APIs.
+  @Deprecated
   public static final Context.Key</*@Nullable*/ Span> CONTEXT_SPAN_KEY =
-      Context.key("opencensus-trace-span-key");
+      Context.<Span>key("opencensus-trace-span-key");
```

Comment:
Default value for context span key is still `null` for backwards-compatibility.

Reply:
Who needs it? gRPC does not need to be null.

Reply:
gRPC expected this behavior in their tests, e.g https://github.com/grpc/grpc-java/blob/master/core/src/test/java/io/grpc/internal/CensusModulesTest.java#L306. I'm not sure if this is used anywhere else. /cc @zhangkun83

Reply:
But we don't change the behavior for the key, only for the new API which we can implement whatever we want.